### ISSUE TITLE
Integrate OneScan for 2D barcode scanning

### ISSUE THREAD

**Issue by vincent-karuri** [onaio/rdt-standard#262]
* [x] Customize JSON form barcode widget to extract information from Zebra OneScan Application
* [x] Autoselect RDT type based on barcode information (this information is not part of the barcode data so won't implement)
* [ ] Implement expiration date confirmation from 2D barcode
* [x] Implement test type confirmation from 2D barcode (this information is not part of the barcode data so won't implement)
* [x] Save barcode information in OpenSRP

**Comment by pld** [comment]
Note that we want to implement this so that

1. there is an interface for the app to read information from an external source
2. our link to OneScan reader implements this interface
3. the current QR code reader will implement the interface
    - NB: for now we don't care about this, we can move the code to the appropriate file, add a TODO to rewrite for the interace, and leave it non-functional

That's basically it, if we wanted to read this data from NFC in the future, we'd be able to do that with another implementation of this interface.

Does that make sense? Happy to have an implementation call before we start on this if needed

**Comment by vincent-karuri** [comment]
We might need to discuss this further.

The idea was to use an intent to launch the OneScan reader and get data from it.

We use the same approach with the QR code reader. The reader is packaged as a form widget, which we'll need to do with the OneScan reader as well.

**Comment by pld** [comment]
Sounds good, after release 1 is done we can discuss in a meeting about 1.5 CC @shivenbhatt1
